### PR TITLE
Add removeAbovePercentile and removeBelowPercentile functions

### PR DESCRIPTION
**Pull request by stivenbb**
Native implementation of [removeAbovePercentile()](http://graphite.readthedocs.io/en/latest/functions.html#graphite.render.functions.removeAbovePercentile) and removeBelowPercentile()](http://graphite.readthedocs.io/en/latest/functions.html#graphite.render.functions.removeBelowPercentile) Graphite functions.

### CODE REVIEW

File: docs/graphite.md [hunk 1]
```diff
@@ -47,7 +47,9 @@ Here are the currently included functions:
 | movingAverage(seriesLists, windowSize) seriesList     |             | Unstable   |
 | perSecond(seriesLists) seriesList                     |             | Stable     |
 | rangeOfSeries(seriesList) series                      |             | Stable     |
-| scale(seriesList, num) series                         |             | Stable     |
+| removeAbovePercentile(seriesList, n) seriesList       |             | Stable     |
+| removeBelowPercentile(seriesList, n) seriesList       |             | Stable     |
+| scale(seriesLists, num) series                        |             | Stable     |
```
Comment (Dieterbe) [inline docs/graphite.md]: why the change to scale? graphite's scale only takes a single serieslist. do we allow multiple? (it's possible, would need to dig into the code)

File: expr/func_removeabovebelowpercentile.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+func (s *FuncRemoveAboveBelowPercentile) Exec(cache map[Req][]models.Series) ([]models.Series, error) {
+	series, err := s.in.Exec(cache)
+	if err != nil {
+		return nil, err
+	}
+
+	if s.n <= 0 {
+		return nil, errors.New("The requested percent is required to be greater than 0")
+	}
```
Comment (Dieterbe) [inline expr/func_removeabovebelowpercentile.go]: should be a validator on the arg as returned by signature. that way we can check it in advance, instead of waiting until we've done a bunch of other work like fetching the data.

File: expr/func_removeabovebelowpercentile.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+		serie.Datapoints = out
+		output = append(output, serie)
+	}
+
+	cache[Req{}] = append(cache[Req{}], output...)
+
+	return output, nil
+}
+
+func getPercentileValue(datapoints []schema.Point, n float64, sortedDatapointVals []float64) float64 {
```
Comment (Dieterbe) [inline expr/func_removeabovebelowpercentile.go]: this function should be documented better. what is sortedDatapointVals ? looks a slice for this function to use as it pleases. also what's the expected/valid range for n? is it `0 < n <= 100` ?

File: expr/func_removeabovebelowpercentile.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	if s.n <= 0 {
+		return nil, errors.New("The requested percent is required to be greater than 0")
+	}
+
+	if len(series) == 0 {
+		return series, nil
+	}
+
+	var output []models.Series
+	sortedDatapointVals := make([]float64, 0, len(series[0].Datapoints)) //reuse float64 slice
```
Comment (Dieterbe) [inline expr/func_removeabovebelowpercentile.go]: confusing comment. maybe "will be reused for each getPercentileValue call" ?

File: docs/graphite.md (hunk 1 above)
Comment (stivenbb) [reply docs/graphite.md]: Sorry this is merge weirdness. It used to say `seriesLists` incorrectly, and I fixed it in one of my commits, but when I rebased this branch, it pulled the old value.

Comment (Dieterbe) [reply docs/graphite.md]: ok

**Comment by stivenbb** [comment]
rebased on top of master

File: expr/func_removeabovebelowpercentile.go [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	sortedDatapointVals = sortedDatapointVals[:0]
+	for _, p := range datapoints {
+		if !math.IsNaN(p.Val) {
+			sortedDatapointVals = append(sortedDatapointVals, p.Val)
+		}
+	}
+
+	sort.Float64s(sortedDatapointVals)
+
+	index := math.Min(math.Ceil(n/100.0*float64(len(sortedDatapointVals)+1)), float64(len(sortedDatapointVals))) - 1
```
Comment (Dieterbe) [inline expr/func_removeabovebelowpercentile.go]: does this correspond to the method used by graphite? it looks different.

Comment (stivenbb) [reply expr/func_removeabovebelowpercentile.go]: Yes, I just simplified it.
